### ISSUE TITLE
File `no-restricted-component-names` - SyntaxError: Unexpected token '.'

### ISSUE THREAD

**Issue by Giildo** [vuejs/eslint-plugin-vue#2245]
**Checklist**

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- eslint: 8.45.0
- eslint-plugin-vue: 9.15.1
- node: 18.17.0
- Debian: 11

**Please show your full configuration:**

```json5
require('@rushstack/eslint-patch/modern-module-resolution');

module.exports = {
  root: true,
  env: {
    node: true,
  },
  extends: [
    'plugin:vue/vue3-essential',
    'eslint:recommended',
    '@vue/eslint-config-typescript',
  ],
  parserOptions: {
    ecmaVersion: 'latest',
  },
  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
    '@typescript-eslint/camelcase': 'off',
  },
};
```

**What actually happened?**
I've this error in the console :

```bash
SyntaxError: Failed to load plugin 'vue' declared in '.eslintrc.js': Unexpected token '.'
Referenced from: /apps/jojotique/new_jojotique/front/.eslintrc.js

/apps/jojotique/new_jojotique/front/node_modules/.pnpm/eslint-plugin-vue@9.15.1_eslint@8.45.0/node_modules/eslint-plugin-vue/lib/rules/no-restricted-component-names.js:119
        const componentName = property.value.value?.toString()
                                                   ^

SyntaxError: Unexpected token '.'
    at wrapSafe (internal/modules/cjs/loader.js:915:16)
    at Module._compile (internal/modules/cjs/loader.js:963:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/apps/jojotique/new_jojotique/front/node_modules/.pnpm/eslint-plugin-vue@9.15.1_eslint@8.45.0/node_modules/eslint-plugin-vue/lib/index.js:122:38)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
Process finished with exit code -1
```

**Comment by FloEdelmann** [comment]
This sounds very similar to https://github.com/vuejs/eslint-plugin-vue/issues/2233, does that help maybe? Optional chaining should be supported on Node v18.

**Comment by Giildo** [comment]
Yes! This https://github.com/vuejs/eslint-plugin-vue/issues/2233#issuecomment-1632089842 is the solution for this issue. 

Thx @FloEdelmann
